website/static/js/share/results.js

```diff
@@ -1,126 +1,252 @@
 'use strict';
 
+var pd = require('pretty-data').pd;
 var $ = require('jquery');
 var m = require('mithril');
 var $osf = require('js/osfHelpers');
 var utils = require('./utils');
-var Results = {};
+var Results = {
+    view: function(ctrl, params) {
+        var vm = params.vm;
+        var resultViews = $.map(vm.results || [], function(result, i) {
+            return m.component(Result, {result: result, vm: vm});
+        });
 
 
-Results.view = function(ctrl) {
-    var res = [];
-    var len = 0;
-    if (ctrl.vm.results){
-        res = $.map(ctrl.vm.results, ctrl.renderResult);
-        len = ctrl.vm.results.length;
-    }
-    return m('', [
-        m('.row', m('.col-md-12', (!utils.arrayEqual(res, [])) ? res : (!ctrl.vm.resultsLoading() && (ctrl.vm.results !== null)) ? m('span', {style: {margin: 'auto'}}, 'No results for this query') : [])),
-        m('.row', m('.col-md-12', ctrl.vm.resultsLoading() ? utils.loadingIcon : [])),
-        m('.row', m('.col-md-12', m('div', {style: {display: 'block', margin: 'auto', 'text-align': 'center'}},
-            len > 0 && len < ctrl.vm.count ?
-            m('a.btn.btn-md.btn-default', {
-                onclick: function(){
-                    utils.loadMore(ctrl.vm)
-                        .then(function(data) {
-                            utils.updateVM(ctrl.vm, data);
-                        });
-                }
-            }, 'More') : [])
-         ))
-    ]);
 
-};
+        var len = 0;
+        if (vm.results){
+            len = vm.results.length;
+        }
+        return m('', [
+            m('.row', m('.col-md-12', (!utils.arrayEqual(resultViews, [])) ? resultViews : (!vm.resultsLoading() && (vm.results !== null)) ? m('span', {style: {margin: 'auto'}}, 'No results for this query') : [])),
+            m('.row', m('.col-md-12', vm.resultsLoading() ? utils.loadingIcon : [])),
+            m('.row', m('.col-md-12', m('div', {style: {display: 'block', margin: 'auto', 'text-align': 'center'}},
+                len > 0 && len < vm.count ?
+                m('a.btn.btn-md.btn-default', {
```

Comment:
When clicked, the user should not be taken to another part of the page; the user should not be scrolled back to the top of the page, for example.